lambdas/ssm-parameters/src/ssm-parameters-handler.ts

```diff
@@ -0,0 +1,131 @@
+import { SessionItem } from "../../../lib/src/types/common-types";
+import { LambdaInterface } from "@aws-lambda-powertools/commons/types";
+import { getParametersByName } from "@aws-lambda-powertools/parameters/ssm";
+import { LogHelper } from "../../../lib/src/Logging/log-helper";
+import { Context } from "aws-lambda";
+import {
+  HandlerMetricExport,
+  MetricsProbe,
+} from "../../../lib/src/Service/metrics-probe";
+
+import {
+  HandlerMetric,
+  CompletionStatus,
+} from "../../../lib/src/MetricTypes/handler-metric-types";
+
+import { MetricUnit } from "@aws-lambda-powertools/metrics";
+import { StrategyUtil, Strategy } from "./types/strategy";
+
+const cacheMinMinutes =
+  Number(process.env.CONFIG_SERVICE_SSM_OPTIMIZED_CACHE_AGE_MIN_MINUTES) || 5;
+const cacheMaxMinutes =
+  Number(process.env.CONFIG_SERVICE_SSM_OPTIMIZED_CACHE_AGE_MAX_MINUTES) || 15;
+
+const logHelper = new LogHelper("SsmParametersHandler");
+
+export class SsmParametersHandler implements LambdaInterface {
+  private metricsProbe: MetricsProbe;
+
+  private randomMaxAgeInSeconds: number;
+
+  constructor(metricsProbe: MetricsProbe) {
+    this.metricsProbe = metricsProbe;
+
+    // Must calulate in seconds
+    const randomValue = Math.random(); // NOSONAR
+    this.randomMaxAgeInSeconds = Math.floor(
```

Comment:
Why are we making max age a random value here? Surely we want it to be a controlled length of time/value?

Reply:
When scaling, there will be multiple lambdas in play. We do not want them all expiring the parameter cache at the same time. In the scenario where we have high traffic and lots of scaled ssm lambdas when these expired we would hit the SSM Rate limits as the caches expire at the same time - the random offset avoid thats.

This is a direct implementation of the same behaviour of the Java SSM caching - 
https://github.com/govuk-one-login/ipv-cri-lib/blob/main/src/main/java/uk/gov/di/ipv/cri/common/library/util/ClientProviderFactory.java#L152